client/my-sites/plans-grid/hooks/npm-ready/data-store/use-grid-plans.tsx

```diff
@@ -323,6 +326,7 @@ const useGridPlans = ( {
 		planSlugs: availablePlanSlugs,
 		storageAddOns,
 	} );
+	const checkoutLink = useAddOnCheckoutLink();
```

Comment:
> hence not return a callback at all

Actually, this is probably off the wrong assumption, sorry. You are running the callback through a loop below. To avoid the callback you'd probably need to pass an array of add-ons and return an index. So no need, returning callback sounds fine (like you have it now). unrelated though to passing in the `selectedSiteSlug`, which sould uncouple the hook from the Calypso dependency :-)

Reply:
>  unrelated though to passing in the selectedSiteSlug, which sould uncouple the hook from the Calypso dependency :-)

Yeah totally agree here. It's a good nudge to kick off our migration to a package with this hook. I'll create an immediate follow up and link it in this PR 😎